hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/operations/RetrieveAndDisposeResultOperation.java

```diff
@@ -44,6 +46,10 @@ public void run() throws Exception {
 
     @Override
     public Object getResponse() {
+        if (result instanceof Throwable) {
```

Comment:
Why is the exception not immediately raised? Why is it stored in 'result'?

Reply:
The task is offloaded, so the result is async. Therefore, the Exception happens on another thread.